proto/spacemesh/v1/tx_types.proto

```diff
@@ -53,3 +53,33 @@ message TransactionState {
     }
     TransactionState state = 2;
 }
+
+// TransactionResultsRequest request object for results stream.
+message TransactionResultsRequest {
+    // id is filter by transaction id.
+    bytes id = 1;
+    // address is a filter by account address, it could be principal or any affected address.
+    bytes address = 2;
+    // start streaming from this layer. if 0 - stream will start from genesis.
+    uint32 start = 3;
```

Comment:
no, it means that you will get results starting from the `start` layer

Reply:
But only in future right? I.e., not for historical layers

Reply:
no, it queries historical data

Reply:
This is a major change to API design. Please see the description here: https://github.com/spacemeshos/api/blob/master/README.md#types-of-endpoints

What's the advantage of streaming historical data, vs. querying?

Not opposed but let's please discuss - and README will need to be updated if we make this change.

Reply:
i don't know what for. stream doesn't return all requested data at once, how much we read is bounded by the stream buffer. so if client is slow - code in spacemesh will wait.

Reply:
I'm asking about _query_ not _stream_

Reply:
during query part db lazily iterates over records and sends them to the grpc stream one by one

Reply:
I'm talking about _query_ endpoints not _stream_ endpoints, i.e., endpoints that do not have a grpc stream. E.g., the methods in MeshService, like this one:

https://github.com/spacemeshos/go-spacemesh/blob/2e58ca2817765c4f7e7c2c9f7968a76d643ce285/api/grpcserver/mesh_service.go#L374-L423

These are slow and memory-inefficient. Some of them include an "offset" i.e. pagination which, as you point out above, is broken. It may not be an issue for this tx service (not totally sure) but it is for other services. What should we do with these?

Reply:
paginated endpoints should have some limit. in this case there will be problems with memory if end is significantly larger than start

i don't know what should be done with paginated endpoints in general, i would probably split them up and replace with object streams:
- activations
- ballots
- txs results
- blocks
- accounts

and more epmeheral streams:
- layer status, only small header (start, hare, tortoise)
- mempool transactions (to track where tx was included)
- proposals

Reply:
okay - let's move this to a new issue - thanks